src/odemis/dataio/tiff.py

```diff
@@ -2613,3 +2624,121 @@ def _iterDirectories(tiff_file):
             tiff_file.ReadDirectory()
             dir_index += 1
             yield dir_index
+
+
+ 
+### FEI Metadata parsing
+def _parseINIMetadataAsDict(metadata: str, delimeter: str = "\r\n"):
+    """
+    Parses ini metadata as dictionary
+    metadata (str): metadata string as ini
+    delimeter (str): delimeter between rows (optional, default: "\r\n")
+    return (dict): metadata as dictionary
+    """
+    md = {}
+    for item in metadata.split(delimeter):
+        if item == "":
+            continue # skip blank lines
+        elif re.match(r"\[(.*?)\]", item):
+            subsystem =  item[1:-1] # rm brackets
+            md[subsystem] = {}
+        else:
+            k, v = item.split("=")   
+            md[subsystem][k] = v
+    return md
+
+
+def _convertFEItoOdemisMD(metadata):
+    """
+    Converts FEI metadata to odemis metadata format
+    metadata (str): metadata string as ini (FEI format)
+    return (dict): metadata as dictionary (odemis format)
+    """
+
+    # parse metadata as dictionary
+    fei_md = _parseINIMetadataAsDict(metadata)
+
+    # convert to odemis format
+    md = {}
+    
+    # general
+    datetime_str = fei_md["User"]["Date"] + " " + fei_md["User"]["Time"]
+    date = datetime.strptime(datetime_str, "%m/%d/%Y %I:%M:%S %p")
+    md[model.MD_ACQ_DATE] = date.timestamp()
+    
+    md[model.MD_HW_NAME] = fei_md["System"]["SystemType"]+"-"+fei_md["System"]["Dnumber"]    
+    md[model.MD_SW_VERSION] = fei_md["System"]["Software"]
+    md[model.MD_PIXEL_SIZE] = (float(fei_md["Scan"]["PixelWidth"]), 
+                                  float(fei_md["Scan"]["PixelHeight"]))
+    
+    md[model.MD_DWELL_TIME] = float(fei_md["Scan"]["Dwelltime"])
+    md[model.MD_DIMS] = "YX"
+    md[model.MD_POS] = (-float(fei_md["Stage"]["StageX"]), 
+                           -float(fei_md["Stage"]["StageY"]),
+                        #    -float(fei_md["Stage"]["StageZ"])
```

Comment:
Normally the MD_POS (and MD_PIXEL_SIZE) metadata should have a 3rd dimension if the dataset has a "Z" dimension. So probably you never need it here (as I don't think users will read "Z stacks" from a TFS SEM, at least in the short/mid term).

Reply:
Initially, I had MD_POS as 3D, but it isn't compatible with tiling which requires 2D POS, so I made it 2D. Alternatively I can change the transform to support mismatched dimensions (only take YX). 

Error from weave with 3D MD_POS

```bash
Exception has occurred: ValueError
matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 3 is different from 2)
  File "/home/odemis/development/odemis/src/odemis/util/transform.py", line 969, in inverse
    translation = -numpy.matmul(matrix, self.translation)
  File "/home/odemis/development/odemis/src/odemis/util/img.py", line 1012, in rotate_img_metadata
    p1 = transl.inverse().apply(pos)
  File "/home/odemis/development/odemis/src/odemis/acq/stitching/_weaver.py", line 79, in getFullImage
    tiles.append(img.rotate_img_metadata(tile, -rotation, center_of_rot))
  File "/home/odemis/development/odemis/src/odemis/acq/stitching/_simple.py", line 111, in weave
    stitched_image = weaver.getFullImage()
  File "/home/odemis/development/odemis/src/odemis/util/dataio.py", line 252, in _import_images_and_stitch
    stitched_image = weave(images)
  File "/home/odemis/development/odemis/dev/main.py", line 31, in main
    stitched_image = _import_images_and_stitch(filenames)
  File "/home/odemis/development/odemis/dev/main.py", line 45, in <module>
    main()
ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 3 is different from 2)
```